app/src/main/java/com/tavarus/artabletop/MainActivity.kt

```diff
@@ -1,121 +1,69 @@
-
-        signInButton.setOnClickListener {
-            val providers = arrayListOf(
-                AuthUI.IdpConfig.EmailBuilder().build(),
-                AuthUI.IdpConfig.GoogleBuilder().build())
-            startActivityForResult(
-                AuthUI.getInstance()
-                    .createSignInIntentBuilder()
-                    .setAvailableProviders(providers)
-                    .build(), RC_SIGN_IN)
-        }
-        //Eventually this would be a full menu
-        settingsButton.setOnClickListener {
-            AuthUI.getInstance().signOut(this)
-                .addOnCompleteListener { unauthedUI() }
-        }
-
-        if (FirebaseAuth.getInstance().currentUser == null) {
-            unauthedUI()
+        if (!AuthController.INSTANCE.isLoggedIn()) {
+            addFragment(LoginFragment())
         } else {
-            authedUI()
-            fetchUser()
+            setupBoard { addFragment(BoardFragment()) }
         }
     }
 
-
-    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
-        super.onActivityResult(requestCode, resultCode, data)
-
-        if (requestCode == RC_SIGN_IN) {
-            val response = IdpResponse.fromResultIntent(data)
-            if (resultCode == Activity.RESULT_OK) {
-                authedUI()
-                fetchUser()
-            } else {
-                // Sign in failed. If response is null the user canceled the
-                // sign-in flow using the back button. Otherwise check
-                // response.getError().getErrorCode() and handle the error.
-                // ...
-            }
-        }
+    fun onLogin() {
+       setupBoard { replaceFragment(BoardFragment()) }
     }
 
-    fun authedUI() {
-        arFragment!!.planeDiscoveryController.show()
-        signInButton.visibility = View.GONE
-        settingsButton.visibility = View.VISIBLE
-        //It would be nice to see if we could add back the default planeDiscoveryController InstructionView
-        //(That's the little phone that moves around)
-        arFragment!!.setOnTapArPlaneListener { hitResult: HitResult, _: Plane, _: MotionEvent ->
-            // Create the Anchor.
-            val anchor = hitResult.createAnchor()
-            val anchorNode = AnchorNode(anchor)
-            anchorNode.setParent(arFragment!!.arSceneView.scene)
-            // Create the board and add it to the anchor.
-            val boardNode = board!!.boardNode()
-            anchorNode.addChild(boardNode)
-        }
-    }
-
-    fun unauthedUI() {
-        arFragment!!.planeDiscoveryController.hide()
-        signInButton.visibility = View.VISIBLE
-        settingsButton.visibility = View.GONE
-        arFragment!!.planeDiscoveryController.setInstructionView(null)
-    }
-
-    fun fetchUser() {
-
+    fun setupBoard(onSuccess: () -> Unit) {
         val firestore = FirebaseFirestore.getInstance()
         val user = FirebaseAuth.getInstance().currentUser
```

Comment:
This can be retrieved from the AuthController, right?

Reply:
What portion do you think should go in the AuthController, and what portion in the BoardController? I should probably move the user into AuthController, but do you think fetching the board should go into BoardController?